### PR TITLE
UI: Dark mode toggle button on Management Server

### PR DESCRIPTION
**Pull request by utchoang**
### Description

Fixes https://github.com/apache/cloudstack/issues/5042

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [x] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### Screenshots (if appropriate):
- Light Mode (default)
image

- Dark Mode
image

- Project Mode
image

### CODE REVIEW

**Comment by rhtyd** [comment]
@blueorangutan ui

**Comment by rhtyd** [comment]
@utchoang are you doing anymore work or is this ready for testing/review?

**Comment by rhtyd** [comment]
Nevermind I tested looks like it's not functional yet.

**Comment by utchoang** [comment]
@rhtyd 
I'm still working on it and not in review mode yet. Thanks.

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by DaanHoogland** [comment]
site looks good, can't judge the code.

**Comment by davidjumani** [comment]
@utchoang The side navbar text is in black
Screenshot from 2021-06-04 15-10-22

**Comment by rhtyd** [comment]
@utchoang can you refer to https://preview.pro.ant.design/dashboard/analysis?navTheme=realDark&fixSiderbar=true&colorWeak=false&pwa=false and have the implementation similar to this and the spec - https://ant.design/docs/spec/dark

**Comment by utchoang** [comment]
@davidjumani 
Is there any mistake? But I tested it and it's ok
image

**Comment by utchoang** [comment]
@rhtyd 
Do you mean the same as the settings in the pro version?
image

**Comment by davidjumani** [comment]
@blueorangutan ui

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by utchoang** [comment]
Setting Drawer:
image
image

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by utchoang** [comment]
@blueorangutan ui

File: ui/src/assets/icons/light.svg [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+            </feMerge>
+        </filter>
+        <rect id="path-2" x="0" y="0" width="48" height="40" rx="4"></rect>
+        <filter x="-4.2%" y="-2.5%" width="108.3%" height="110.0%" filterUnits="objectBoundingBox" id="filter-4">
+            <feOffset dx="0" dy="1" in="SourceAlpha" result="shadowOffsetOuter1"></feOffset>
+            <feGaussianBlur stdDeviation="0.5" in="shadowOffsetOuter1" result="shadowBlurOuter1"></feGaussianBlur>
+            <feColorMatrix values="0 0 0 0 0   0 0 0 0 0   0 0 0 0 0  0 0 0 0.1 0" type="matrix" in="shadowBlurOuter1"></feColorMatrix>
+        </filter>
+    </defs>
+    <g id="配置面板" stroke="none" stroke-width="1" fill="none" fill-rule="evenodd">
```
Comment (GabrielBrascher) [inline ui/src/assets/icons/light.svg:23]: Is this (id="配置面板") necessary?
If possible changing to something in English would be better :slightly_smiling_face:.

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by sureshanaparti** [comment]
ping @utchoang can you please resolve the conflicts.

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by nvazquez** [comment]
@utchoang please fix the conflict. Is it ready for review/test?

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by utchoang** [comment]
@nvazquez 

> @utchoang please fix the conflict. Is it ready for review/test?

Yes. It ready for review

**Comment by nvazquez** [comment]
Looks great @utchoang! One suggestion: can you please fix the text color of the selected item in the sidebar when the dark mode is enabled? Its not legible
image

**Comment by utchoang** [comment]
@blueorangutan ui

**Review by nvazquez — APPROVED** [review]
Thanks @utchoang LGTM - manually tested

**Review by GabrielBrascher — APPROVED** [review]
LGTM based on manual tests. 
Thanks @utchoang!

**Comment by nvazquez** [comment]
Ping @rhtyd @davidjumani @shwstppr @Pearl1594 this is a very nice enhancement, can you please review?

**Comment by davidjumani** [comment]
@utchoang In the forms, the placeholder text is not faded (grey) but white like any text the user fills in which can be confusing

**Comment by rhtyd** [comment]
I'm okay to merge if dark mode is not the default and there are no regressions in the normal mode.

**Comment by shwstppr** [comment]
Looks nice!

@utchoang 
1. possible to add a "Reset to default" option in theme panel?
2. In project view, some text color change needed in dark mode
Screenshot from 2021-08-03 11-20-34

**Comment by utchoang** [comment]
@blueorangutan ui

**Review by DaanHoogland — APPROVED** [review]
I like the "reset to default", LGTM

**Review by shwstppr — APPROVED** [review]

**Review by RodrigoDLopez — APPROVED** [review]
Manually tested. LGTM

**Comment by nvazquez** [comment]
Thanks all for testing, we are only missing code review @davidjumani @Pearl1594 can you handle that?

**Comment by rhtyd** [comment]
@davidjumani have you tested it, are you lgtm on this?

**Comment by apurv-1** [comment]
The UI looks awesome @utchoang! 

Pointing some minor bugs, I came across while manually testing UI.
1. Some text fields in forms are turning white on hover(In the dark mode). Some of them are:
   - Instances > Kubernetes > Create Kubernetes Cluster > Node root disk size (in GB) field.
   - Images > Kubernetes ISOs > Add Kubernetes Version >  Min CPU Cores &  Min Memory (in MB) field.
   - Infrastructure > Zone > Add Zone > Under Zone details most of the DNS fields.
   - Service Offerings > Compute & System Offerings > Add Compute Offering > most of the fields.
 
https://user-images.githubusercontent.com/53378650/128712900-b671f152-235f-417a-9c4b-889c139354c2.mp4

2. I don't know if its the issue on my machine but while choosing the theme colour, It works great if I wanted to select a colour on click but while holding the cursor and then selecting it while moving the mouse it lags(maybe because its changing colour and rendering the theme colour at the same time). 

https://user-images.githubusercontent.com/53378650/128715776-5a493042-6170-42f3-9e03-d90566785bbe.mp4

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by utchoang** [comment]
@apurv-1 Can you please check case 1 again? Also for case 2, changing the text color, the background now will only apply when the lost focus from the color picker. Thanks.

**Comment by apurv-1** [comment]
> @apurv-1 Can you please check case 1 again? Also for case 2, changing the text color, the background now will only apply when the lost focus from the color picker. Thanks.

Great, both the changes are fixed & looks really good. Thanks!

**Comment by nvazquez** [comment]
@utchoang another minor issue on the dark mode: when the list view is empty, the 'No data' text is not visible

image

**Comment by utchoang** [comment]
@blueorangutan ui

**Comment by nvazquez** [comment]
Thanks for the quick fix @utchoang, can you fix the latest conflict?

**Comment by utchoang** [comment]
@blueorangutan ui

**Pull request merged**